File: das_backend/src/main/java/ch/sbb/backend/logging/infrastructure/rest/SplunkRequest.java
package ch.sbb.backend.logging.infrastructure.rest;

import com.fasterxml.jackson.annotation.JsonInclude;
import java.time.OffsetDateTime;
import java.util.Map;

@JsonInclude(JsonInclude.Include.NON_NULL)
record SplunkRequest(
    String event,
    Map<String, String> fields,
    String host,
    String index,
    String source,
    OffsetDateTime time,
    String sourcetype
) {

    SplunkRequest(String event, Map<String, String> fields, String source, OffsetDateTime time) {
        this(event, fields, null, null, source, time, null);
    }
}

File: das_backend/src/main/kotlin/ch/sbb/backend/logging/infrastructure/rest/SplunkRequest.kt
package ch.sbb.backend.logging.infrastructure.rest

import com.fasterxml.jackson.annotation.JsonInclude
import java.time.OffsetDateTime

data class SplunkRequest(
    val event: String,
    val fields: Map<String, String>,
    @JsonInclude(JsonInclude.Include.NON_NULL)
    val host: String? = null,
    @JsonInclude(JsonInclude.Include.NON_NULL)
    val index: String? = null,
    val source: String,
    val time: OffsetDateTime,
    @JsonInclude(JsonInclude.Include.NON_NULL)
    val sourcetype: String? = null
)
